src/main/java/Save.java

```diff
@@ -0,0 +1,81 @@
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class Save {
+
+    public static String filePath;
+
+    public static void createSaveFile() throws IOException {
+        File file = new File(filePath);
+        if (!file.exists()) {
+            file.createNewFile();
+            file.mkdirs();
+        }
+    }
+
+    public static void loadSaveFile(ArrayList<Task> tasks) {
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] args = line.split("\\|", -1);
+                if (line.startsWith("T")) {
+                    ToDo todo = new ToDo(args[2]);
+                    if (args[2].equals("X")) {
+                        todo.markAsDone();
+                    }
+                    tasks.add(todo);
+                } else if (line.startsWith("D")) {
+                    Deadline deadline = new Deadline(args[2], args[3]);
+                    if (args[2].equals("X")) {
+                        deadline.markAsDone();
+                    }
+                    tasks.add(deadline);
+                } else {
+                    Event event = new Event(args[2], args[3], args[4]);
+                    if (args[2].equals("X")) {
+                        event.markAsDone();
+                    }
+                    tasks.add(event);
+                }
+            }
+        } catch (IOException exception) {
```

Comment:
the catch only belongs to the file read. once the file is read, rest of the operation can go ahead without troubles.
reordering the `try ... catch`  will eliminate the arrowhead code.